pennylane/templates/subroutines/adder.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
```

Comment:
Instead of having the files in /templates/subroutines, it would be better to have a /templates/arithmetic. (Tagging @soranjh  for confirmation)

Reply:
Let's keep as is. We will have one final PR for documentation.